python/create_service_diagnostics_bundle.py

```diff
@@ -138,99 +52,259 @@ def parse_args() -> dict:
+
+    @property
+    def cluster_name(self) -> str:
+        if self._cluster_name is not None:
+            self._cluster_name = self.get_cluster_name()
+
+        return self._cluster_name
 
-    print("\nWill create bundle for:")
-    print("  Package:         {}".format(args.get("package_name")))
-    print("  Package version: {}".format(args.get("package_version")))
-    print("  Service name:    {}".format(args.get("service_name")))
-    print("  DC/OS version:   {}".format(args.get("dcos_version")))
-    print("  Cluster URL:     {}".format(args.get("cluster_url")))
+    @property
+    def clusters(self) -> list:
+        if self._clusters is None:
+            self._clusters = self.get_attached_clusters()
 
-    if args.get("should_prompt_user"):
-        answer = input("\nProceed? [Y/n]: ")
-        if answer.strip().lower() not in ["yes", "y", ""]:
-            return 0
+        return self._clusters
 
-    rc, _ = FullBundle(
-        args.get("package_name"), args.get("service_name"), args.get("bundles_directory")
-    ).create()
+    @property
+    def marathon_info(self):
+        if self._marathon_app is None:
+            self._marathon_app = self.get_marathon_app(self._service_name)
+            self.validate_package_name(
+                self._package_name, self._marathon_app, self.service_name)
 
-    return rc
+        return self._marathon_app
+
+    @property
+    def package_name(self) -> str:
+        if self._package_name is None:
+            self.load_args()
+
+        return self._package_name
+
+    @property
+    def service_name(self) -> str:
+        if self._service_name is None:
+            self.load_args()
+
+        return self._service_name
+
+    @property
+    def bundles_directory(self) -> str:
+        if self._bundles_directory is None:
+            self.load_args()
+
+        return self._bundles_directory
+
+    @property
+    def should_prompt_user(self) -> bool:
+        if self._should_prompt_user is None:
+            self.load_args()
+
+        return self._should_prompt_user
+
+    @property
+    def package_version(self) -> str:
+        if self._package_version is None:
+            marathon_app = self.marathon_info
+            self._package_version = "n/a"
+            if marathon_app is not None:
+                self._package_version = marathon_app.get("labels", {}).get(
+                    "DCOS_PACKAGE_VERSION")
+
+        return self._package_version
+
+    @property
+    def dcos_version(self) -> str:
+        if self._dcos_version is None:
+            self._dcos_version = self.clusters[0]["version"]
+        return self._dcos_version
+
+    @property
+    def cluster_url(self) -> str:
+        if self._cluster_url is None:
+            self._cluster_url = self.clusters[0]["url"]
+        return self._cluster_url
+
+    @property
+    def service_diagnostics_version(self) -> str:
+        if self._service_diagnostics_version is None:
+            __location__ = os.path.realpath(
```

Comment:
What's special about this variable that it needs to be dunderized?

Reply:
changed